### PR TITLE
Initial Version release v1.0.0

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/viralmetagenome/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/viralmetagenome _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by fmobegi — COMMENTED** [review]
workflow viralmetagenome.nf
1. Redundant ch_multiqc_config definition (lines 93 and 448)
2. Redundant ch_multiqc_logo definition (lines 95 and 453)
3. Remove unused function `noContigSamplesToMultiQC`
4. The parameter initialization section has very complex conditional logic (It works!). Could these benefit from having helper functions?
5. Having the `ch_` prefix on all channel variables makes it easier to distinguish them from regular variables for such a complex pipeline. Please check throughout the workflow.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -8,8 +8,7 @@ body:
         Before you post this issue, please check the documentation:
 
         - [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
-        - [Viralgenie pipeline documentation](https://joon-klaps.github.io/viralgenie/latest/usage)
-
+        - [nf-core/viralmetagenome pipeline documentation](https://nf-co.re/viralmetagenome/usage)
```
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply .github/ISSUE_TEMPLATE/bug_report.yml:11]: Yeah there are still quite a few differences between the master and dev. The master was still published under viralgenie and dev under `nf-core/viralmetagenome`. Hence the need for an actual release.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
--   [ ] Make sure your code lints (`nf-core lint`).
--   [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
--   [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
--   [ ] Usage Documentation in `docs/usage.md` is updated.
--   [ ] Output Documentation in `docs/output.md` is updated.
--   [ ] `CHANGELOG.md` is updated.
--   [ ] `README.md` is updated (including new tool citations and authors/contributors).
+- [ ] This comment contains a description of changes (with reason).
+- [ ] If you've fixed a bug or added code that should be tested, add tests!
+- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/viralmetagenome/tree/master/.github/CONTRIBUTING.md)
```
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: Same here, I removed the contributing file in the viralgenie, but reincluded it in nf-core/viralmetagenome

File: README.md [hunk 3]
```diff
@@ -1,61 +1,58 @@
...
 
 1. Read QC ([`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/))
 2. Performs optional read pre-processing
-    - Adapter trimming([`fastp`](https://github.com/OpenGene/fastp), [`Trimmomatic`](https://github.com/usadellab/Trimmomatic))
-    - Read UMI deduplication ([`HUMID`](https://humid.readthedocs.io/en/latest/usage.html))
-    - Low complexity and quality filtering ([`bbduk`](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/), [`prinseq++`](https://github.com/Adrian-Cantu/PRINSEQ-plus-plus))
-    - Host-read removal ([`BowTie2`](http://bowtie-bio.sourceforge.net/bowtie2/))
+   - Adapter trimming([`fastp`](https://github.com/OpenGene/fastp), [`Trimmomatic`](https://github.com/usadellab/Trimmomatic))
+   - Read UMI deduplication ([`HUMID`](https://humid.readthedocs.io/en/latest/usage.html))
+   - [Optional] Combining of reads (CAT)
```
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply README.md:31]: Yeah it is, it's the step afer fastp & trimmomatic. Note, that the dev branch of the github page is where the most recent version is.

File: conf/tests/test_group.config [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        cpus: 4,
+        memory: '15.GB',
+        time: '1.h'
+    ]
+}
+
+
+params {
+    config_profile_name        = 'Test group profile'
+    config_profile_description = 'Minimal test dataset to check pipeline function when combining samples of the same group'
```
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply conf/tests/test_group.config:24]: No, the same original sample. The taxprofiler equivalents would then be: 
Sample = group
runacession = sample

However, I didn't follow Taxprofilers schema but viralrecon's in order to keep the samplesheets compatible. As I'm guessing most of my users come from there.

File: conf/tests/test_umi.config [hunk 5]
```diff
@@ -22,26 +22,25 @@ params {
     config_profile_description = 'UMI test dataset to check pipeline function for reads with unique molecular identifiers (UMIs)'
 
     // Input data
-    input     = "${projectDir}/assets/samplesheets/samplesheet.csv"
+    input     = params.pipelines_testdata_base_path + 'viralmetagenome/samplesheets/samplesheet.csv'
     outdir    = "testing"
```
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply conf/tests/test_umi.config:26]: I guess not, but it easier for me not to type it out everytime i test the pipeline locally.

[the remark this answers is not in the document]
Comment (sofstam) [reply conf/tests/test_umi.config]: Ok!

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sofstam) [reply README.md]: I missed that, thank you for explaining

**Review by sofstam — COMMENTED** [review]
I left a few more comments in this PR.

File: conf/test.config [hunk 6]
```diff
@@ -18,28 +18,26 @@ process {
...
     // Input data
-    input     = "${projectDir}/assets/samplesheets/samplesheet.csv"
+    input     = params.pipelines_testdata_base_path + 'viralmetagenome/samplesheets/samplesheet.csv'
     outdir    = "testing"
-    metadata  = "${projectDir}/assets/samplesheets/metadata_test.tsv"
+    metadata  = params.pipelines_testdata_base_path + 'viralmetagenome/samplesheets/metadata_test.tsv'
 
     skip_complexity_filtering   = false
     trim_tool                   = 'fastp'
     assemblers                  = 'spades,megahit'
```
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply conf/test.config:32]: test was to have a fast (complete) runthrough of the pipeline. Test_full, is for the complete pipeline functionality and the other test_profiles are for edge cases.

File: assets/multiqc_config.yml
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply assets/multiqc_config.yml:1]: Like a complete html file?

File: conf/test.config (hunk 6 above)
[the remark this answers is not in the document]
Comment (sofstam) [reply conf/test.config:32]: Sounds good, maybe this can also be documented so that it is clear for the users?

File: assets/multiqc_config.yml
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply assets/multiqc_config.yml:1]: Not yet, I'll generate one.

**Review by sofstam — APPROVED** [review]
All the comments have been adressed. Great work 🚀
